cli/ceph/osd/crush.py

```diff
@@ -0,0 +1,162 @@
+        """
+        cmd = f"{self.base_cmd} rule"
+        for arg in args:
+            cmd += f" {arg}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def set_device_class(self, device_class, osd_id):
+        """
+        To set device class to osd
+        Args:
+            device_class (str): device class (hdd/ssd)
+            osd_id (list): list of osd's
+        """
+        cmd = f"{self.base_cmd} set-device-class {device_class}"
+        for _osd in osd_id:
+            cmd += f" {_osd}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def rm_device_class(self, device_class, osd_id):
+        """
+        To remove device class to osd
+        Args:
+            device_class (str): device class (hdd/ssd)
+            osd_id (list): list of osd's
+        """
+        cmd = f"{self.base_cmd} rm-device-class {device_class}"
+        for _osd in osd_id:
+            cmd += f" {_osd}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def rename_device_class(self, old_name, new_name):
+        """
+        To rename device class
+        Args:
+            old_name (str): old class name
+            new_name (str): new class name
+        """
+        cmd = f"{self.base_cmd} class rename {old_name} {new_name}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def ls_osd(self, device_class):
+        """
+        To list all OSDs that belong to a particular class
+        Args:
+            device_class (str): device class (hdd/ssd)
+        """
+        cmd = f"{self.base_cmd} class ls-osd {device_class}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def add_bucket(self, name, type):
+        """
+        To add a bucket instance to CRUSH hierarchy
+        Args:
+            name (str): bucket name
+            type (str): type of bucket
+        """
+        cmd = f"{self.base_cmd} add-bucket {name} {type}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def move_bucket(self, name, type):
+        """
+        To move a bucket instance to a particular location in CRUSH hierarchy
+        Args:
+            name (str): bucket name
+            type (str): type of bucket
+        """
+        cmd = f"{self.base_cmd} move {name} {type}"
+        out = self.execute(sudo=True, cmd=cmd)
+        if isinstance(out, tuple):
+            return out[0].strip()
+        return out
+
+    def remove_bucket(self, name):
```

Comment:
Same command is mentioned below can't it used ?

Reply:
One is specific to remove bucket instance in CRUSH hierarchy and the other is to remove osd from CRUSH map of a running cluster. So its better to have both in my opinion.

Reply:
I don't see any change in code. It's duplicate method.

Reply:
combined both to single method